app/grants/router.py

```diff
@@ -88,59 +93,161 @@ def get_queryset(self):
-        """Generate Grants report for an ethereum address"""
+    @ratelimit(key='ip', rate='5/s')
+    def contributions_rec_report(self, request):
+        """Genrate Grantee Report for an Grant"""
 
-        grants_queryset = Grant.objects.all()
-        contributions_queryset = Contribution.objects.all()
+        results_limit = 30
 
-        grantee_serializer = GranteeSerializer
-        donor_serializer = DonorSerializer
+        page = self.request.query_params.get('page', '1')
+        if not page or not page.isdigit():
+            page = 1
 
-        param_keys = self.request.query_params.keys()
-        if 'eth_address' not in param_keys:
-            raise NotFound(detail='Missing required parameter: eth_address')
-        eth_address = self.request.query_params.get('eth_address')
-
-        # Filter Grants and Contributions by the given eth_address
-        grants_queryset = grants_queryset.filter(admin_address=eth_address)
-        contributions_queryset = contributions_queryset.filter(subscription__contributor_address=eth_address)
-
-        # Filter Grantee info by from_timestamp
-        format = '%Y-%m-%dT%H:%M:%SZ'
-        if 'from_timestamp' in param_keys:
-            from_timestamp = self.request.query_params.get('from_timestamp')
+        grant_pk = self.request.query_params.get('id')
+        if not grant_pk or not grant_pk.isdigit():
+            return Response({
+                'error': 'missing manadatory parameter id'
+            })
+
+        now = datetime.now()
+        to_timestamp = self.request.query_params.get('to_timestamp')
+        from_timestamp = self.request.query_params.get('from_timestamp')
+
+        # Check timestamp
+        if not from_timestamp:
+            from_timestamp = now - timedelta(days=30)
+        else:
             try:
                 from_timestamp = datetime.strptime(from_timestamp, format)
             except ValueError:
-                raise NotFound(detail="Please provide from_timestamp in the format: "+format)
-        else:
-            from_timestamp = datetime(1, 1, 1, 0, 0)
-        grants_queryset = grants_queryset.filter(subscriptions__subscription_contribution__created_on__gte=from_timestamp)
+                return Response({
+                    'error': 'from_timestamp is not in format YYYY-MM-DD'
+                })
 
-        # Filter Grantee info by to_timestamp
-        if 'to_timestamp' in param_keys:
-            to_timestamp = self.request.query_params.get('to_timestamp')
+        if not to_timestamp:
+            to_timestamp = now
+        else:
             try:
                 to_timestamp = datetime.strptime(to_timestamp, format)
             except ValueError:
-                raise NotFound(detail="Please provide to_timestamp in the format: "+format)
+                return Response({
+                    'error': 'to_timestamp is is not in the format YYYY-MM-DD'
+                })
+
+        if (to_timestamp - from_timestamp).days > 32:
```

Comment:
This wouldn't catch the value `== 32`. Should be `>= 32` or `> 31`.

Reply:
Why are we using 32 days as the marker here? Wouldn't 31 (max days in a month) make more sense?

Reply:
I can tweak it to `>= 31` but aren't they the same ?